File: lihua-vue/src/api/monitor/server/type/server-info.ts
import type {CpuMonitor} from "@/api/monitor/server/type/cpu-monitor.ts";
import type {MemoryMonitor} from "@/api/monitor/server/type/memory-monitor.ts";
import type {JvmMonitor} from "@/api/monitor/server/type/jvm-monitor.ts";
import type {DiskMonitor} from "@/api/monitor/server/type/disk-monitor.ts";

export interface ServerInfo {
    cpuMonitor: CpuMonitor;
    memoryMonitor: MemoryMonitor;
    jvmMonitor: JvmMonitor;
    diskMonitor: DiskMonitor;
}

File: lihua-biz/lihua-monitor/src/main/java/com/lihua/monitor/model/ServerInfo.java
package com.lihua.monitor.model;

import lombok.Data;
import lombok.experimental.Accessors;

/**
 * 服务器信息
 */
@Data
@Accessors(chain = true)
public class ServerInfo {

    /**
     * cpu信息
     */
    private CpuMonitor cpuMonitor;

    /**
     * 内存信息
     */
    private MemoryMonitor memoryMonitor;

    /**
     * java虚拟机信息
     */
    private JvmMonitor jvmMonitor;

    /**
     * 服务器硬盘信息
     */
    private DiskMonitor diskMonitor;

}
